lib/config/normalization.js

```diff
@@ -187,6 +187,7 @@ const getNormalizedWebpackOptions = config => {
 		externals: config.externals,
 		externalsPresets: cloneObject(config.externalsPresets),
 		externalsType: config.externalsType,
+		ignoreBrowserWarnings: config.ignoreBrowserWarnings,
```

Comment:
Let's put it in MF plugin options, not as global

Reply:
What do you want to do about `ShareRuntimeModule` or any other potential browser console warns? Didn't realize until looking now that this one is called from `RuntimePlugin` from `WebpackOptionsApply` from `webpack:createCompiler`

Reply:
I see, in this case move them to `output.generateBrowserWarnings: boolean`, because it is not the top level option, also please simplify runtime logic, if you don't need warning, you don't need extra code:
```
`var warn = ${runtimeTemplate.basicFunction("msg", [
				'if (typeof console === "undefined" || !console.warn) return;',
				`var ignoreWarnings = ${
					this.compilation.options.ignoreBrowserWarnings &&
					`[${this.compilation.options.ignoreBrowserWarnings.toString()}]`
				};`,
				`var ignoreWarning = !ignoreWarnings ? false : ignoreWarnings.some(${runtimeTemplate.basicFunction(
					"ignore",
					["return ignore.test(msg);"]
				)});`,
				"!ignoreWarning && console.warn(msg);"
			])};`,
```

look how many code we generate and it is do nothing when you want to ignore warnings, also we always run `getInvalidVersionMessage`, but why we need it if we don't output this?

Reply:
This was following a similar pattern to the `ignoreWarnings` option. So it needs to check the message to know whether to ignore it or not. But if we're fine with just using a boolean option, agree this is simpler.

Reply:
Yeah, I think boolean is more better here, because it looks strange to ignore some problems...

Reply:
These test cases with the expectWarning factory are failing now...it's sending the output of console.warn to the console instead of the mocked console.warn. Tested this locally and it runs correctly.

Reply:
I see that the issue for testing was the functions that call `console.warn` work, but it registers the actual console instead of the mock console for testing.